rules.d/99-systemd.rules.in

```diff
@@ -19,7 +19,7 @@ SUBSYSTEM=="block", ACTION=="add", ENV{DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1", E
 
 # Ignore encrypted devices with no identified superblock on it, since
 # we are probably still calling mke2fs or mkswap on it.
-SUBSYSTEM=="block", ENV{DM_UUID}=="CRYPT-*", ENV{ID_PART_TABLE_TYPE}=="", ENV{ID_FS_USAGE}=="", ENV{SYSTEMD_READY}="0"
+SUBSYSTEM=="block", ENV{DM_UUID}=="CRYPT-*", ENV{ID_PART_TABLE_TYPE}=="", ENV{ID_FS_USAGE}=="", ENV{DM_SUSPENDED}=="0", ENV{SYSTEMD_READY}="0"
```

Comment:
hmm, this doesn't look right?

the status quoa ante was that crypt devices without a superblock should not be considered ready.

With your change you change this to allow crypt services lacking a superblock but which are suspended to be considered ready, and that doesn't appear to me is what you are trying to do.

Reply:
I want to prevent systemd from unmounting FS on top of suspended devices. This is mostly for case of CHANGE events (i.e. device was there previously and now is suspended and CHANGE was triggered). ADD events are handled on the line above, as suspended devices have `DM_UDEV_DISABLE_OTHER_RULES_FLAG}=="1"` so they won't be marked as ready.

Btw, with the suspended device you don't know whether it was superblock or not because we won't read it.

Reply:
Specifically, I want systemd to not react to bellow sequence of commands,

```
dmsetup suspend /dev/mapper/disk
udevadm trigger
dmsetup resume /dev/mapper/disk
```

Now this sequence of step will cause unmount of FS that is on top of /dev/mapper/disk (if this is configured through fstab). This is not wanted and unexpected. After suspend every attempt to use the device will block in kernel anyway.  There is no risk of data corruption and no need for systemd to spawn unmount.

Reply:
I probably shouldn't have said "...w/o superblock info" in the commit message. We do not have superblock info not because it is not there but because we skip scanning the device because it is suspended.

Reply:
hmm, oh.

so this patch will address that specific issue, but create a new issue.

I mean, we don't want to access the device at all when it is suspended. Which the existing rule enforced. So the logic so far was "if it's suspended, then it's not really there".

But with your change you want that what is already set up isn#t destroyed if the device is suspended. So the logic you are looking for is "if it's suspended, then it's no really gone".

The fact is that our current model cannot express this concept of "hey, if a device is suspended then don't do anything new with it, but also don't destroy everything old on it".

This kinda suggests that while DM suspend is in effect on a block device we should try to inherit the SYSTEMD_READY state from the previous database, no?

Reply:
Hmm, what about actually reacting or rather not reacting to events at all if DM_SUSPENDED key is set to 1. I mean SYSTEMD_READY is about state changes but for suspended devices we probably shouldn't do any state transitions until the device is resumed because that we are getting in the event are incomplete.

Reply:
yeah, but that's precisely what inheriting SYSTEMD_READY from the previous db state will give you